libpromises/evalfunction.c

```diff
@@ -318,7 +320,39 @@ static JsonElement* VarRefValueToJson(const EvalContext *ctx, const FnCall *fp,
                         switch (var->rval.type)
                         {
                         case RVAL_TYPE_SCALAR:
-                            JsonObjectAppendString(holder, last_key, var->rval.item);
+                            if (JsonGetElementType(holder) != JSON_ELEMENT_TYPE_CONTAINER)
+                            {
+                                Log(LOG_LEVEL_WARNING,
+                                    "Replacing a non-container JSON element '%s' with a new empty container"
+                                    " (for the '%s' subkey)",
+                                    JsonGetPropertyAsString(holder), last_key);
+
+                                assert(holder_parent != NULL);
+
+                                JsonElement *empty_container = JsonObjectCreate(10);
+                                char *element_name = xstrdup(JsonGetPropertyAsString(holder));
+                                JsonObjectAppendElement(holder_parent,
+                                                        element_name,
+                                                        empty_container);
+                                free (element_name);
```

Comment:
Why allocate this string when JsonObjectAppendElement uses xstrdup anyway?

Reply:
Comment added.

Reply:
Thanks!